src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      currentState = { ...currentState, ...actions[i].extraData };
```

Comment:
The use of spread syntax here is correct for merging properties. Ensure that `actions[i].extraData` is always an object to prevent runtime errors.